api/base/api_globals.py

```diff
@@ -0,0 +1,10 @@
+"""Global variables related to Django requests
+
+Made available in a separate file so as to be importable by Flask code, before (or in the absence of) Django
+"""
+import threading
+
+api_globals = threading.local()
```

Comment:
That's an interesting question; I'm not deeply familiar with `LocalProxy`. Short answer: I'd have to pass the buck to someone who's used it hands-on. 

It seems that there might be advantages if we're using other threading models... at the tradeoff of some hidden assumptions and behaviors that may not be applicable to Django. (eg, see the ClosingIterator docstring in werkzeug; Django has its own layer of middleware that should work as long as a thread is only handling one request at once)